compiler/rustc_error_codes/src/error_codes/E0756.md

```diff
@@ -0,0 +1,29 @@
+The `ffi_const` attribute was used on something else than a foreign function
+declaration.
+
+Erroneous code example:
+
+```compile_fail,E0756
+#![feature(ffi_const)]
+
+#[ffi_const] // error!
+pub fn foo() {}
+# fn main() {}
+```
+
+The `ffi_const` attribute can only be used on foreign function declarations
+which have no effects except for their return value:
```

Comment:
Let's not confuse people into thinking about effect type systems ;)
```suggestion
which have no side effects except for their return value:
```

Reply:
Previously I was suggesting
```
The `ffi_pure` attribute can only be used on foreign functions which do not have
side effects or infinite loops:
```
which could be used exactly the same here, I wonder if that is better since I combined multiple parts from the book.

Reply:
C assumes that all functions don't have infinite loops; it's called "forward progress guarantee". The rules around it are somewhat odd and I don't think they belong here, see https://blog.rust-lang.org/inside-rust/2020/03/19/terminating-rust.html for details.